packages/browser-agent-core/src/common/util/submit-data.js

```diff
@@ -53,5 +53,8 @@ submitData.img = function img (url) {
  * Do NOT call this function outside of a guaranteed web window environment.
  */
 submitData.beacon = function (url, body) {
-  return window.navigator.sendBeacon(url, body)
+  // Navigator has to be bound to ensure it does not error in some browsers
+  // https://xgwang.me/posts/you-may-not-know-beacon/#it-may-throw-error%2C-be-sure-to-catch
+  const send = globalScope.navigator.sendBeacon.bind(navigator)
```

Comment:
(minor note: `navigator` cannot be reassigned & ought to be safe as is)
<img width="832" alt="image" src="https://user-images.githubusercontent.com/28714891/213580724-fd0164c9-f15f-4582-9a6a-f6efb6bcf706.png">

Reply:
Update: actually can you just remove import/use of `globalScope`? That module is meant for APIs interchangeable between window and workers. In this case, the `navigator.sendBeacon` is only relevant to the window, so it should just be `window.navigator.sendBeacon.bind(navigator)`.